### PR TITLE
Additional Docker tips

### LINKED ISSUE #809: Consolidate best practices in Docker section of troubleshooting document
The Docker section in [puppeteer/docs/troubleshooting.md](https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md#running-puppeteer-in-docker) suggests some undesirable flags. Instead, please consider recommending better solutions found by the community.

- [x] https://github.com/jessfraz/dockerfiles/blob/master/chrome/stable/Dockerfile shows how to set up a dedicated `chrome` user, so sandboxing can remain on
- [x]  https://github.com/c0b/chrome-in-docker/issues/1 documents that setting up a larger `/dev/shm` size can help eliminate crashes when rendering large pages.
  * Current fix: in the `docker run` add `--shm-size=1gb`

Thank you!

### PR DESCRIPTION
**Pull request by ebidel**
Fixes #809

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -145,11 +145,25 @@ docker build -t puppeteer-chrome-linux .
...
+
+By default, Docker runs a container with a `/dev/shm` shared memory space 6MB.
+This is [typically too small](https://github.com/c0b/chrome-in-docker/issues/1) for Chrome
+and will cause Chrome to crash when rendering large pages. To fix, run the container
+with `docker run --shm-size=1gb` to increase the size of `/dev/shm`. In the future,
+this won't be necessary. See [crbug.com/736452](https://bugs.chromium.org/p/chromium/issues/detail?id=736452).
+
+If you're seeing other weird errors when launching Chrome, try running the container
+with `docker run --cap-add=SYS_ADMIN` when developing locally. Since the Dockerfile
+adds a `pptr` user as a non-privileged user, it may not have all the necessary privileges.
```
Comment (aslushnikov) [inline docs/troubleshooting.md:166]: Let's add a reference to [dumb-init](..) here - it turned out there's a special treatment for the processes with PID=1, which makes it hard to terminate chrome properly.

Discussed here: https://github.com/GoogleChrome/puppeteer/issues/615#issuecomment-330355210

**Review by aslushnikov — APPROVED** [review]

**Review by CesarLanderos — APPROVED** [review]
This is great!, it took me a couple of days to get to the bottom of this issue a month ago (chrome not running properly on docker) and this is exactly what i did!

<3

File: docs/troubleshooting.md (hunk 1 above)
Comment (ebidel) [reply docs/troubleshooting.md:166]: done

**Comment by ebidel** [comment]
@aslushnikov not sure why travis is unhappy. Looks like there's errors on `node utils/doclint/cli.js`? That known?

**Comment by aslushnikov** [comment]
@ebidel should be fixed in https://github.com/GoogleChrome/puppeteer/commit/a467d35a5f1dbfa9ae91015c164ad2f786265aee

**Comment by joelgriffith** [comment]
Might also be a good idea to call out fonts as well. See a lot of issues regarding fonts/unicode characters either not rendering or showing up in a bizarre fashion.

For instance, running:
```
const browser = await puppeteer.launch();
const page = await browser.newPage();
await page.goto('http://eosrei.github.io/emojione-color-font/full-demo.html');
await page.screenshot({path: 'screenshot.png'});
await browser.close();
```
On https://try-puppeteer.appspot.com/, results in:
26dbc732-a9b2-47f6-8f12-5ee5bb592cc6

**Comment by ebidel** [comment]
Thanks. Going to merge this then.

**Pull request merged**